crates/utils/src/settings/structs.rs

```diff
@@ -1,29 +1,32 @@
 use doku::Document;
```

Comment:
Right now these dont use Option, so it requires changes across the project. I gave it a try, but in federation code it gets pretty ugly, youd have to unwrap the settings.federation option because its simply required. So i feel like it would end up uglier than the current code. If you still think it makes sense, best try to change it yourself.